cmd/asset-syncer/utils.go

```diff
@@ -254,6 +255,15 @@ func doReq(url string, headers map[string]string) ([]byte, error) {
 	}
 
 	if res.StatusCode != http.StatusOK {
+		if err == nil {
```

Comment:
This is unrelated but I noticed that when getting an error, the message was in the body of the response so I am extracting it. I will add a unit test for this.

Reply:
I'm not sure what this condition here is doing given that 5 lines above we're already returning if the error is non-nil (ie. I'd expect that it is always nil by line 258 and so this is superfluous - perhaps left over from an earlier edit?)